pkgs/development/libraries/faac/default.nix

```diff
@@ -1,20 +1,31 @@
-{ stdenv, fetchurl, mp4v2 }:
+{ stdenv, fetchurl
+, mp4v2Support ? true, mp4v2 ? null
+, drmSupport ? false # Digital Radio Mondiale
+}:
+
+assert mp4v2Support -> (mp4v2 != null);
 
 stdenv.mkDerivation rec {
-  name = "faac-1.28";
+  name = "faac-${version}";
+  version = "1.28";
 
   src = fetchurl {
     url = "mirror://sourceforge/faac/${name}.tar.gz";
     sha256 = "1pqr7nf6p2r283n0yby2czd3iy159gz8rfinkis7vcfgyjci2565";
   };
 
-  buildInputs = [ mp4v2 ];
+  configureFlags = []
+    ++ stdenv.lib.optional mp4v2Support "--with-mp4v2"
+    ++ stdenv.lib.optional drmSupport "--enable-drm";
+
+  buildInputs = []
+    ++ stdenv.lib.optional mp4v2Support mp4v2;
 
-  meta = {
+  meta = with stdenv.lib; {
     description = "Open source MPEG-4 and MPEG-2 AAC encoder";
-    homepage = http://www.audiocoding.com/faac.html;
-    # Incompatible with GPL. Some changes to the base code, included in faac,
-    # are under LGPL though.
-    license = stdenv.lib.licenses.unfreeRedistributable;
+    homepage    = http://www.audiocoding.com/faac.html;
+    license     = stdenv.lib.licenses.unfree;
```

Comment:
@wmertens The best I can tell is that regardless of patents, other distros do build faac as non-free software, so I will revert my licensing change.  Thanks for bringing it to my attention.